client/state/selectors/can-any-site-connect-domains.ts

```diff
@@ -0,0 +1,26 @@
+import { createSelector } from '@automattic/state-utils';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import getSites from 'calypso/state/selectors/get-sites';
+import type { AppState } from 'calypso/types';
+
+import 'calypso/state/ui/init';
+
+/**
+ * Return true if a user has sites that are able to be connected to domains.
+ * Returns false when a user has no sites or only self hosted Jetpack sites.
+ *
+ * @param state Global state tree
+ */
+
+export const canAnySiteConnectDomains = createSelector(
+	( state: AppState ): boolean =>
+		getSites( state ).some(
+			( site ) =>
+				site &&
+				canCurrentUser( state, site.ID, 'manage_options' ) &&
+				! ( site.jetpack && ! ( site?.options?.is_automated_transfer ?? false ) ) && // Simple and Atomic sites. Not Jetpack sites.
+				! ( site?.is_wpcom_staging_site ?? false ) &&
+				! ( site?.options?.is_domain_only ?? false )
```

Comment:
What's the is_wpcom_staging_site line checking? Double negations are hard :smile: 

Is this a correct reading:

> If we have a site and we can manage its options, we can connect the site to a domain
> Unless: 
> - It's a jetpack that is also not an atomic site or
> - It's a wpcom staging site or
> - It's a domain only site

Can wpcom staging sites not be connected to domains?  (aren't these just atomic sites?)

Reply:
Looks like these are something new? As in not referring to sites with a wpcomstaging.com subdomain (atomic sites).

Reply:
![Screenshot 2023-07-25 at 12-41-48 Hosting Configuration ‹ Site Title — WordPress com](https://github.com/Automattic/wp-calypso/assets/811776/c51e3de5-e225-4e6b-b270-e8b31febb952)

Nice!

Reply:
They are atomic sites, but they're not able to be mapped to domains so this looks good :ship:'n it